### PR TITLE
Clean Up of Table/Block

### PR DESCRIPTION
**Pull request by mprammer2**
Changes focus on readability, clean up, and inline'ing of relevant methods. Significant documentation via doc-strings has been added. Tests were slightly modified to match the new behavior of "InsertRecord" and similar methods, which have all been unified to return the same output.

### CODE REVIEW

**Review by srsuryadev — APPROVED** [review]
Nice work!

File: src/storage/block.cc [hunk 1]
```diff
@@ -384,48 +283,40 @@ void Block::insert_records(
...
+      block_map[row_id] = {blockInfo.block_id, row};
+      this->InsertRecords(sliced_column_data);
     } else {
        reduced_count++;
     }
   }
+  return num_rows - 1;
 }
 
 // Note that this funciton assumes that the valid column is in column_data
```
Comment (johnchronis) [inline src/storage/block.cc:303]: funciton -> function here and in rest of file

File: src/storage/tests/table_test.cc [hunk 2]
```diff
@@ -100,99 +100,77 @@ class HustleTableTest : public testing::Test {
...
   auto table_from_file = read_from_file("table.hsl");
-
-  EXPECT_TRUE(table_from_file->get_block(0)->get_valid_column()->Equals(
-      *table.get_block(0)->get_valid_column()));
-  EXPECT_TRUE(table_from_file->get_block(0)->get_records()->Equals(
-      *table.get_block(0)->get_records()));
-  EXPECT_TRUE(table_from_file->get_block(1)->get_records()->Equals(
-      *table.get_block(1)->get_records()));
+  EXPECT_TRUE(
+      table_from_file->get_block(0)->get_valid_column()->Equals(*table.get_block(0)->get_valid_column()));
```
Comment (johnchronis) [inline src/storage/tests/table_test.cc:156]: Is this <=80 chars? it looks wider, Github should have a line at 80 chars to make this easy to check :)

File: src/storage/tests/block_test.cc [hunk 3]
```diff
@@ -235,12 +233,11 @@ TEST_F(HustleBlockTest, FullBlock) {
...
   // With 1 KB block size, we can store 8 copies of the first record
   for (int i = 0; i < 8; i++) {
-    block.insert_record((uint8_t *)record_string_1.data(), byte_widths_1);
+    block.InsertRecord((uint8_t *)record_string_1.data(), byte_widths_1);
   }
 
   // This block cannot hold a 9th copy of the first record
-  int row_index =
-      block.insert_record((uint8_t *)record_string_1.data(), byte_widths_1);
+  int row_index = block.InsertRecord((uint8_t *)record_string_1.data(), byte_widths_1);
```
Comment (johnchronis) [inline src/storage/tests/block_test.cc:240]: This looks wider than 80 chars

**Review by johnchronis — APPROVED** [review]

**Comment by mprammer2** [comment]
Next commit adds a ".clang-format" file to the root of the project, which was generated via:
`clang-format -style=google -dump-config > .clang-format`
This can be imported into CLion in settings via:
`Editor > Code Style > Enable ClangFormat`
Though I'm not sure how to add it to Visual Studio.

Given how we have people with different IDEs working on the project, I think that it's a good idea to look into a multi-IDE automated formatting system. I've solved the line-spacing issues by running this this clang format. If we like this style we can start to deploy it elsewhere in the project.

**Pull request merged**